src/core/providers/memory/qgsmemoryprovider.cpp

```diff
@@ -104,6 +104,7 @@ QgsMemoryProvider::QgsMemoryProvider( const QString &uri, const ProviderOptions
 
                   // boolean
                   << QgsVectorDataProvider::NativeType( tr( "Boolean" ), QStringLiteral( "bool" ), QVariant::Bool )
+                  << QgsVectorDataProvider::NativeType( tr( "Boolean" ), QStringLiteral( "boolean" ), QVariant::Bool )
```

Comment:
this shouldn't be repeated

Reply:
Ok, than just the one with `boolean` must be kept, because of how the URI is parsed at line 183.
Corrected in last commit.